lib/linter-ruby.coffee

```diff
@@ -1,33 +1,54 @@
-linterPath = atom.packages.getLoadedPackage("linter").path
-Linter = require "#{linterPath}/lib/linter"
-
-class LinterRuby extends Linter
-  # The syntax that the linter handles. May be a string or
-  # list/tuple of strings. Names should be all lowercase.
-  @syntax: ['source.ruby', 'source.ruby.rails', 'source.ruby.rspec']
-
-  # A string, list, tuple or callable that returns a string, list or tuple,
-  # containing the command line (with arguments) used to lint.
-  cmd: 'ruby -wc'
-
-  executablePath: null
-
-  errorStream: 'stderr'
-
-  linterName: 'ruby'
-
-  # A regex pattern used to extract information from the executable's output.
-  regex:
-    '.+:(?<line>\\d+):((?<warning> warning:)|(?<error>))(?<message>.+)'
-
-  constructor: (editor)->
-    super(editor)
-
-    @rubyLinter = atom.config.observe 'linter-ruby.rubyExecutablePath', =>
-      @executablePath = atom.config.get 'linter-ruby.rubyExecutablePath'
-
-  destroy: ->
-    super
-    @rubyLinter.dispose()
-
-module.exports = LinterRuby
+child_process = require 'child_process'
+path = require 'path'
+
+module.exports = LinterRuby =
+  config:
+    executablePath:
+      default: 'ruby'
+      type: 'string'
+
+  activate: ->
+    unless atom.packages.getLoadedPackages 'linter-plus'
+      atom.notifications.addError(
+        '[linter-ruby] `linter-plus` package not found, please install it')
+
+  provideLinter: -> {
+    grammarScopes: ['source.ruby', 'source.ruby.rails', 'source.ruby.rspec']
+    scope: 'file'
+    lint: @lint
+    lintOnFly: false
+  }
+
+  lint: (TextEditor) ->
+    regex = ///
+      (.+): #File with issue
+      (\d+):\s+ #Line with issue
+      (\w+(?:\s(?:error|warning))?) #Issue type
+      (?:,\s) #A command and space seperating the type and message
+      (.+) #Message explaining the issue
+    ///
+    return new Promise (Resolve) ->
+      filePath = TextEditor.getPath()
+      if filePath # Files that have not be saved
+        file = path.basename(filePath)
+        cwd = path.dirname(filePath)
+        cmd = atom.config.get "linter-ruby.executablePath"
+        cmd = "#{cmd} -wc"
+        cmd = "#{cmd} #{file}"
+        console.log "linter-ruby command: #{cmd}" if atom.inDevMode()
+        data = []
+        process = child_process.exec cmd, {cwd: cwd}
+        process.stderr.on 'data', (d) -> data.push d.toString()
+        process.on 'close', ->
+          toReturn = []
+          for line in data
+            console.log "linter-ruby output: #{line}" if atom.inDevMode()
+            if line.match regex
+              [file, line, type, message] = line.match(regex)[1..4]
+              toReturn.push(
+                type: type,
```

Comment:
I see. So maybe it would be better to have raw type instead of `'Info'`

```coffee
type: if type is 'syntax error' then 'Error' else if type is 'warning' then 'Warning' else type
```

and still have the coloring

![screen shot 2015-08-06 at 9 35 59](https://cloud.githubusercontent.com/assets/1788571/9105256/cb1413f8-3c1e-11e5-8eca-a51fc95336ea.png)
![screen shot 2015-08-06 at 9 35 21](https://cloud.githubusercontent.com/assets/1788571/9105257/cb149f30-3c1e-11e5-88e6-268b01ef235b.png)